 framework/webapp/config/fop.xconf | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git framework/webapp/config/fop.xconf framework/webapp/config/fop.xconf
index 27576e37a5..36cdaa57c2 100644
--- framework/webapp/config/fop.xconf
+++ framework/webapp/config/fop.xconf
@@ -32,7 +32,7 @@ the location of this file.
 
   <!-- Base URL for resolving relative URLs -->
   <!-- We have to keep HTTP URL , FOP does not support HTTPS baseUrl for external graphics due to certificate issue (not remember exact reason) -->
-  <base>http://localhost:8080</base>
+  <base>https://demo-stable.ofbiz.apache.org</base>
 
   <!-- Source resolution in dpi (dots/pixels per inch) for determining the size of pixels in SVG and bitmap images, default: 72dpi -->
   <source-resolution>72</source-resolution>
